File: 21553 - Sai Nandan D/Week 10/45_Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        
        if n == 1: return 0
        
        jump_no = 1
        l, r = 1, nums[0]

        while r < n-1:
            jump_no += 1     
            r_nxt = max(i + nums[i] for i in range(l, r+1))
            l, r = r+1, r_nxt            

        return jump_no

File: 21561 - Suvam Tamang/Week 10/45. Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int n=nums.length;
        int [] min_dist = new int[n];
        for(int i=1; i<n; i++) {
            min_dist[i] = Integer.MAX_VALUE;
        }
        min_dist[0] = 0;
        for(int i=0; i<n; i++) {
            for(int j=i; j<=i+nums[i] && j<n; j++) {
                min_dist[j] = Math.min(min_dist[j], min_dist[i]+1);
            }
        }
        return min_dist[n-1];
    }
}
